leader/leader.go

```diff
@@ -182,6 +183,14 @@ func Become(ctx context.Context, lockName string, opts ...Option) error {
 					if err != nil {
 						log.Error(err, "Leader pod could not be deleted.")
 					}
+				case isNotReadyNode(ctx, config.Client, leaderPod.Spec.NodeName):
+					log.Info("the status of the node where operator pod with leader lock was running has been 'notReady'")
+					log.Info("Deleting the leader.")
+
+					//Mark the termainating status to the leaderPod and Delete the configmap lock
+					if err := deleteLeader(ctx, config.Client, leaderPod, existing); err != nil {
```

Comment:
Should we not use the same func across all implementation? Also, why is required to remove the configMap?

Reply:
1) The reason i added the function(delete Leader) is for the lint-test. golangci-lint limits Cyclomatic complexity in the function. so we have to divide the code into multiple functions.
2) if i do not remove the configmap, leader election will not occur imediately when the node where the leader pod is running was 'not ready' status. Because the leader pod was just marked as 'terminating'.(the garbage collector do not delete the configmap until the leader pod is actually deleted)